core/src/test/java/org/springframework/security/authorization/AuthorizationAdvisorProxyFactoryTests.java

```diff
@@ -360,6 +362,19 @@ public void proxyWhenDefaultsThenInstanceOfAuthorizationProxy() {
 		assertThat(target).isSameAs(this.flight);
 	}
 
+	@Test
+	public void asyncProxyWhenDefaultsThenNoException() {
+		AuthorizationAdvisorProxyFactory factory = AuthorizationAdvisorProxyFactory.withDefaults();
+		List<CompletableFuture<Void>> futures = new ArrayList<>();
+
+		for (int i = 0; i < 1000; i++) {
```

Comment:
Testing concurrency is quite tricky as it is non-deterministic. 

Let's instead consider a test in `PrePostMethodSecurityConfiguration` that can confirm that the advisors in the `AuthorizationAdvisorProxyFactory` bean are ordered.